src/server/ua_discovery_mdns_avahi.c

```diff
@@ -881,32 +855,29 @@ UA_Discovery_addRecord(UA_DiscoveryManager *dm, const UA_String servername,
     }
 
     /* [servername]-[hostname] */
-    char serviceDomain[63];
-    createServiceDomain(serviceDomain, 63, servername, hostname);
+    UA_Byte serviceDomainBuf[64];
+    UA_String serviceDomain = {64, serviceDomainBuf};
+    createServiceDomain(&serviceDomain, servername, hostname);
 
-    UA_Boolean exists = UA_Discovery_recordExists(dm, serviceDomain,
-                                                  port, protocol);
+    UA_Boolean exists = UA_Discovery_recordExists(dm, serviceDomain, port, protocol);
     if(exists == true)
         return UA_STATUSCODE_GOOD;
 
     UA_LOG_INFO(dm->sc.server->config.logging, UA_LOGCATEGORY_DISCOVERY,
                 "Multicast DNS: add record for domain: %s", serviceDomain);
```

Comment:
serviceDomain is now an UA String. This will fix the [segmentation fault](https://github.com/open62541/open62541/pull/7231/files#r2054378169) in the format_string function:
```suggestion
                "Multicast DNS: add record for domain: %S", serviceDomain);
```

I'm still testing a couple of things, but so far I haven't seen any other problems.

Reply:
👍